src/pysource/models.py

```diff
@@ -253,6 +299,13 @@ def __init__(self, origin, spacing, shape, m, space_order=2, nbl=40,
         # density
         self._init_density(rho, space_order)
         self._dm = self._gen_phys_param(dm, 'dm', space_order)
+
+        # Additional parameter fields for Viscoacoustic operators
+        if qp is not None:
+            self._is_viscoacoustic = True
+            self.qp = self._gen_phys_param(qp, 'qp', space_order)
```

Comment:
infer abc_type here to avoid that carry around in options which is also unsafe for users